mmaction/datasets/pipelines/augmentations.py

```diff
@@ -44,6 +52,108 @@ def _init_lazy_if_proper(results, lazy):
 
 
 @PIPELINES.register_module()
+class PoseCompact:
+    """Convert the coordinates of keypoints to make it more compact.
+    Specifically, it first find a tight bounding box that surrounds all joints
+    in each frame, then we expand the tight box by a given padding ratio. For
+    example, if 'padding == 0.25', then the expanded box has unchanged center,
+    and 1.25x width and height.
+
+    Args:
+        padding (float): The padding size. Default: 0.25.
+        threshold (int): The threshold for the tight bounding box. If the width
+            or height of the tight bounding box is smaller than the threshold,
+            we do not perform the compact operation. Default: 10.
+        hw_ratio (float | tuple[float] | None): The hw_ratio of the expanded
+            box. Float indicates the specific ratio and tuple indicates a
+            ratio range. If set as None, it means there is no requirement on
+            hw_ratio. Default: None.
+        allow_imgpad (bool): Whether to allow expanding the box outside the
+            image to meet the hw_ratio requirement. Default: True.
+
+    Returns:
+        type: Description of returned object.
+    """
+
+    def __init__(self,
+                 padding=0.25,
+                 threshold=10,
+                 hw_ratio=None,
+                 allow_imgpad=True):
+
+        self.padding = padding
+        self.threshold = threshold
+        if hw_ratio is not None:
+            hw_ratio = _pair(hw_ratio)
+
+        self.hw_ratio = hw_ratio
+
+        self.allow_imgpad = allow_imgpad
+        assert self.padding >= 0
+
+    def __call__(self, results):
+        img_shape = results['img_shape']
+        h, w = img_shape
+        kp = results['kp']
+        min_x, min_y, max_x, max_y = np.Inf, np.Inf, -np.Inf, -np.Inf
+
+        # Make NaN zero
+        kp[np.isnan(kp)] = 0.
+        kp_x = kp[..., 0]
+        kp_y = kp[..., 1]
+
+        # There is at least one legal keypoint
+        if np.sum(kp_x != 0) or np.sum(kp_y != 0):
```

Comment:
This condition might allow errors. If kp_x is all zeros, and kp_y is not, then

![image](https://user-images.githubusercontent.com/9464825/116689466-bb046180-a9ea-11eb-9aa3-74abd6da1434.png)

Reply:
Maybe try

```python
kp =  np.nan_to_num(kp)
kp_x = kp[..., 0]
kp_y = kp[..., 1]
min_x = np.min(kp_x[kp_x != 0], , initial=-np.Inf)
...
```

Reply:
OK, I will modify the code.